EEJT-000A/main/java/task2/SearchingController.java

```diff
@@ -0,0 +1,46 @@
+package task2;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+public class SearchingController {
+    private Parameters parameters;
+    private SearchByParam searchByParam;
+
+    public SearchingController(Parameters parameters) {
+        this.parameters = parameters;
+    }
+
+    public List<String> doChain(List<String> list) {
+        if (!isNull(parameters)) {
```

Comment:
Please avoid using `isNull` method for simple checks. It was intender to be used as method reference (e.g. in streams
Objects::isNull)